src/dataunion/Contracts.ts

```diff
@@ -37,11 +41,15 @@ export class Contracts {
     }
 
     getDataUnionMainnetAddress(dataUnionName: string, deployerAddress: EthereumAddress) {
-        if (!this.factoryMainnetAddress) {
-            throw new Error('StreamrClient has no factoryMainnetAddress configuration.')
+        if (!isAddress(this.factoryMainnetAddress)) {
+            throw new Error('StreamrClient factoryMainnetAddress configuration is ' + this.factoryMainnetAddress ? 'missing' : 'not a valid Ethereum address')
+        }
+
+        if (!isAddress(this.templateMainnetAddress)) {
+            throw new Error('StreamrClient dataUnionTemplateMainnetAddress configuration is ' + this.templateMainnetAddress ? 'missing' : 'not a valid Ethereum address')
```

Comment:
I think this ternary is around the wrong way, 

```suggestion
            throw new Error('StreamrClient dataUnionTemplateMainnetAddress configuration is ' + this.templateMainnetAddress ? 'not a valid Ethereum address' : 'missing')
```